config/manager/manager.yaml

```diff
@@ -22,17 +22,35 @@ spec:
       labels:
         control-plane: controller-manager
     spec:
+      securityContext:
+        runAsNonRoot: true
       containers:
-      - name: manager
-        image: controller:latest
-        # imagePullPolicy: IfNotPresent
-        command:
+      - command:
         - /manager
+        args:
+        - --leader-elect
+        image: controller:latest
+        name: manager
+        securityContext:
+          allowPrivilegeEscalation: false
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: 8081
+          initialDelaySeconds: 15
+          periodSeconds: 20
+        readinessProbe:
+          httpGet:
+            path: /readyz
+            port: 8081
+          initialDelaySeconds: 5
+          periodSeconds: 10
         resources:
           limits:
             cpu: 100m
             memory: 30Mi
           requests:
             cpu: 100m
             memory: 20Mi
-      terminationGracePeriodSeconds: 20
+      serviceAccountName: controller-manager
```

Comment:
It may require to update Workload Identity settings written in `README.md`.

Reply:
Thanks, fixed it! I think having a helm-chart would greatly simplify the deployment process here. Currently it requires too many files to be manually created and certain lines have to be searched and modified. I will see if `kpt` can help with that or `helm`.